languages/ruby/spec/oso/polar/data_filtering/new_filter_spec.rb

```diff
@@ -0,0 +1,419 @@
+                kind: 'one',
+                other_type: 'Sign',
+                my_field: 'sign_name',
+                other_field: 'name'
+              )
+            }
+          )
+          subject.register_class(
+            Planet,
+            fields: {
+              name: String,
+              signs: Relation.new(
+                kind: 'many',
+                other_type: 'Sign',
+                my_field: 'name',
+                other_field: 'planet_name'
+              )
+            }
+          )
+          subject.register_class(
+            Sign,
+            fields: {
+              name: String,
+              element: String,
+              planet: Relation.new(
+                kind: 'one',
+                other_type: 'Planet',
+                my_field: 'planet_name',
+                other_field: 'name'
+              ),
+              people: Relation.new(
+                kind: 'many',
+                other_type: 'Person',
+                my_field: 'name',
+                other_field: 'sign_name'
+              )
+            }
+          )
+        end
+        it 'test_authorize_scalar_attribute_eq' do
+          subject.load_str <<~POL
+            allow(_: Person, "read", _: Sign{element: "fire"});
+            allow(_: Person{sign}, "read", sign);
+          POL
+          query = subject.authzd_query(Person.find('sam'), 'read', Sign)
+          expected_signs = %w[pisces aries sagittarius leo].map { |n| Sign.find n }
+          expect(query.to_a).to contain_exactly(*expected_signs)
+        end
+
+        it 'test_authorize_scalar_attribute_condition' do
+          subject.load_str <<~POL
+            # signs ruled by jupiter can read their own people
+            # this rule relies on direct object comparison (aka `field == None`) working correctly :)
+            allow(sign: Sign{planet}, "read", _: Person{sign}) if planet.name = "jupiter";
+            # every sign can read a pisces named sam
+            allow(_: Sign, "read", _: Person {sign, name: "sam"}) if sign.name = "pisces";
+            # earth signs can read people with air signs
+            allow(_: Sign{element: "earth"}, "read", _: Person{sign}) if sign.element = "air";
+          POL
+
+          test = lambda do |person, sign|
+            (person.sign == sign && sign.planet.name == 'jupiter') ||
+              (person.name == 'sam' && person.sign.name == 'pisces') ||
+              (sign.element == 'earth' && person.sign.element == 'air')
+          end
+          Sign.all.each do |sign|
+            expected = Person.all.select do |person|
+              test[person, sign]
+            end
+            query = subject.authzd_query sign, 'read', Person
+            expect(query.to_a).to contain_exactly(*expected)
+          end
+        end
+
+        it 'test_nested_relationship_many_single' do
+          subject.load_str <<~POL
+            allow(_: Person{sign}, _, _: Planet{signs}) if sign in signs;
```

Comment:
This isn't the same test as elsewhere. This is the Django test:

```
        allow(user: test_app2::User, "read", post: test_app2::Post) if
            tag in post.tags and
            tag.created_by = user;
```

Why are all these tests using a different domain than the initial ones (Django, SQLAlchemy)? It makes it harder to compare & port. Noticed that the JS version is also a different domain.

Reply:
(No need to fix this right now at all but I'm just noticing it, going forward we should probably use the same test suite)

Reply:
I'll at least add TODO comments for these

Reply:
Cool thanks